Emby.Server.Implementations/ScheduledTasks/Tasks/AudioNormalizationPostScanTask.cs

```diff
@@ -0,0 +1,198 @@
+using MediaBrowser.Model.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Emby.Server.Implementations.ScheduledTasks.Tasks;
+
+/// <summary>
+/// The splashscreen post scan task.
+/// </summary>
+public partial class AudioNormalizationTask : IScheduledTask
+{
+    private readonly IItemRepository _itemRepository;
+    private readonly ILibraryManager _libraryManager;
+    private readonly IMediaEncoder _mediaEncoder;
+    private readonly IConfigurationManager _configurationManager;
+    private readonly ILocalizationManager _localization;
+    private readonly ILogger<AudioNormalizationTask> _logger;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="AudioNormalizationTask"/> class.
+    /// </summary>
+    /// <param name="itemRepository">Instance of the <see cref="IItemRepository"/> interface.</param>
+    /// <param name="libraryManager">Instance of the <see cref="ILibraryManager"/> interface.</param>
+    /// <param name="mediaEncoder">Instance of the <see cref="IMediaEncoder"/> interface.</param>
+    /// <param name="configurationManager">Instance of the <see cref="IConfigurationManager"/> interface.</param>
+    /// <param name="localizationManager">Instance of the <see cref="ILocalizationManager"/> interface.</param>
+    /// <param name="logger">Instance of the <see cref="ILogger{AudioNormalizationTask}"/> interface.</param>
+    public AudioNormalizationTask(
+        IItemRepository itemRepository,
+        ILibraryManager libraryManager,
+        IMediaEncoder mediaEncoder,
+        IConfigurationManager configurationManager,
+        ILocalizationManager localizationManager,
+        ILogger<AudioNormalizationTask> logger)
+    {
+        _itemRepository = itemRepository;
+        _libraryManager = libraryManager;
+        _mediaEncoder = mediaEncoder;
+        _configurationManager = configurationManager;
+        _localization = localizationManager;
+        _logger = logger;
+    }
+
+    /// <inheritdoc />
+    public string Name => _localization.GetLocalizedString("TaskAudioNormalization");
+
+    /// <inheritdoc />
+    public string Description => _localization.GetLocalizedString("TaskAudioNormalizationDescription");
+
+    /// <inheritdoc />
+    public string Category => _localization.GetLocalizedString("TasksLibraryCategory");
+
+    /// <inheritdoc />
+    public string Key => "AudioNormalization";
+
+    [GeneratedRegex(@"I:\s+(.*?)\s+LUFS")]
+    private static partial Regex LUFSRegex();
+
+    /// <inheritdoc />
+    public async Task ExecuteAsync(IProgress<double> progress, CancellationToken cancellationToken)
+    {
+        foreach (var library in _libraryManager.RootFolder.Children)
+        {
+            var libraryOptions = _libraryManager.GetLibraryOptions(library);
+            if (!libraryOptions.EnableLUFSScan)
+            {
+                continue;
+            }
+
+            // Album gain
+            var albums = _libraryManager.GetItemList(new InternalItemsQuery
```

Comment:
I'm never a fan of pulling all of an item type from the database.

Reply:
Yeah, but this can be changed when progress reporting is added